### ISSUE TITLE
Support Iterator implement macro for Iter

### ISSUE THREAD

**Issue by farodin91** [intellij-rust/intellij-rust#1489]
https://doc.rust-lang.org/std/slice/struct.Iter.html

**Comment by vlad20012** [comment]
working on it

**Comment by vlad20012** [comment]
What impls and for what types we should hardcode besides `Iterator` for Iter & IterMut? Only primitive binops and Fn* types comes to mind.

**Comment by farodin91** [comment]
What do you think with what types we should hardcode?

**Comment by farodin91** [comment]
https://doc.rust-lang.org/nightly/src/core/slice/mod.rs.html#1117-1215
https://doc.rust-lang.org/nightly/std/slice/struct.Iter.html

**Comment by vlad20012** [comment]
I just did it, and I try to remember what other types implemented with macros in std.

**Comment by farodin91** [comment]
I understand

**Comment by farodin91** [comment]
grep macro_rules on std and core

**Comment by farodin91** [comment]
libstd
```
ascii.rs:1220:        macro_rules! test {
ascii.rs:1243:        macro_rules! test {
ascii.rs:1292:    macro_rules! assert_all {
ascii.rs:1313:    macro_rules! assert_none {
ffi/os_str.rs:623:macro_rules! impl_cmp {
os/raw.rs:92:    macro_rules! ok {
net/mod.rs:105:macro_rules! doit {
net/addr.rs:840:        macro_rules! try_opt {
net/udp.rs:708:    macro_rules! t {
net/tcp.rs:805:    macro_rules! t {
sync/mpsc/select.rs:375:    macro_rules! select {
sys/unix/mod.rs:117:macro_rules! impl_is_minus_one {
sys/unix/process/process_common.rs:423:    macro_rules! t {
sys/unix/process/process_unix.rs:147:        macro_rules! t {
sys/unix/ext/net.rs:1351:    macro_rules! or_panic {
sys/unix/weak.rs:36:macro_rules! weak {
sys/redox/net/dns/mod.rs:72:        macro_rules! push_u8 {
sys/redox/net/dns/mod.rs:78:        macro_rules! push_n16 {
sys/redox/net/dns/mod.rs:107:        macro_rules! pop_u8 {
sys/redox/net/dns/mod.rs:119:        macro_rules! pop_n16 {
sys/redox/net/dns/mod.rs:131:        macro_rules! pop_data {
sys/redox/net/dns/mod.rs:146:        macro_rules! pop_name {
sys/redox/process.rs:266:        macro_rules! t {
sys/windows/mod.rs:216:macro_rules! impl_is_zero {
sys/windows/backtrace/mod.rs:35:macro_rules! sym {
sys/windows/net.rs:63:macro_rules! impl_is_minus_one {
sys/windows/compat.rs:48:macro_rules! compat_fn {
fs.rs:1871:    macro_rules! check { ($e:expr) => (
fs.rs:1879:    macro_rules! error { ($e:expr, $s:expr) => (
fs.rs:1888:    macro_rules! error { ($e:expr, $s:expr) => (
time/mod.rs:445:    macro_rules! assert_almost_eq {
sys_common/mod.rs:99:macro_rules! rtabort {
sys_common/backtrace.rs:338:                    macro_rules! demangle {
sys_common/backtrace.rs:394:    macro_rules! t { ($a:expr, $b:expr) => ({
path.rs:2459:macro_rules! impl_cmp {
path.rs:2497:macro_rules! impl_cmp_os_str {
path.rs:2560:    macro_rules! t(
path.rs:3529:        macro_rules! tp(
path.rs:3623:        macro_rules! tp(
path.rs:3683:        macro_rules! tfn(
path.rs:3717:        macro_rules! tfe(
path.rs:3754:        macro_rules! t {
path.rs:3779:        macro_rules! tc(
macros.rs:44:macro_rules! panic {
macros.rs:105:macro_rules! print {
macros.rs:132:macro_rules! println {
macros.rs:153:macro_rules! eprint {
macros.rs:171:macro_rules! eprintln {
macros.rs:214:macro_rules! select {
macros.rs:231:macro_rules! assert_approx_eq {
macros.rs:241:/// These macros do not have any corresponding definition with a `macro_rules!`
macros.rs:254:    macro_rules! compile_error { ($msg:expr) => ({ /* compiler built-in */ }) }
macros.rs:282:    macro_rules! format_args { ($fmt:expr, $($args:tt)*) => ({
macros.rs:303:    macro_rules! env { ($name:expr) => ({ /* compiler built-in */ }) }
macros.rs:323:    macro_rules! option_env { ($name:expr) => ({ /* compiler built-in */ }) }
macros.rs:351:    macro_rules! concat_idents {
macros.rs:372:    macro_rules! concat { ($($e:expr),*) => ({ /* compiler built-in */ }) }
macros.rs:388:    macro_rules! line { () => ({ /* compiler built-in */ }) }
macros.rs:404:    macro_rules! column { () => ({ /* compiler built-in */ }) }
macros.rs:421:    macro_rules! file { () => ({ /* compiler built-in */ }) }
macros.rs:440:    macro_rules! stringify { ($t:tt) => ({ /* compiler built-in */ }) }
macros.rs:457:    macro_rules! include_str { ($file:expr) => ({ /* compiler built-in */ }) }
macros.rs:474:    macro_rules! include_bytes { ($file:expr) => ({ /* compiler built-in */ }) }
macros.rs:495:    macro_rules! module_path { () => ({ /* compiler built-in */ }) }
macros.rs:517:    macro_rules! cfg { ($($cfg:tt)*) => ({ /* compiler built-in */ }) }
macros.rs:555:    macro_rules! include { ($file:expr) => ({ /* compiler built-in */ }) }
num.rs:129:    macro_rules! test_is_power_of_two {
num.rs:150:    macro_rules! test_next_power_of_two {
num.rs:170:    macro_rules! test_checked_next_power_of_two {
num.rs:202:        macro_rules! assert_pow {
thread/local.rs:139:macro_rules! thread_local {
thread/local.rs:162:macro_rules! __thread_local_inner {
thread/local.rs:193:macro_rules! thread_local {
thread/local.rs:229:macro_rules! __thread_local_inner {
```
libcore
```
iter/range.rs:50:macro_rules! step_identical_methods {
iter/range.rs:74:macro_rules! step_impl_unsigned {
iter/range.rs:103:macro_rules! step_impl_signed {
iter/range.rs:145:macro_rules! step_impl_no_between {
iter/range.rs:178:macro_rules! range_exact_iter_impl {
iter/range.rs:185:macro_rules! range_incl_exact_iter_impl {
iter/range.rs:194:macro_rules! range_trusted_len_impl {
iter/range.rs:201:macro_rules! range_incl_trusted_len_impl {
iter/traits.rs:663:macro_rules! integer_sum_product {
iter/traits.rs:703:macro_rules! float_sum_product {
clone.rs:140:macro_rules! clone_impl {
default.rs:128:macro_rules! default_impl {
str/mod.rs:779:macro_rules! derive_pattern_clone {
str/mod.rs:830:macro_rules! generate_pattern_iterators {
str/mod.rs:1394:        macro_rules! err {
str/mod.rs:1403:        macro_rules! next { () => {{
str/pattern.rs:349:macro_rules! pattern_methods {
str/pattern.rs:377:macro_rules! searcher_methods {
slice/mod.rs:230:macro_rules! slice_offset {
slice/mod.rs:242:macro_rules! make_ref {
slice/mod.rs:255:macro_rules! make_ref_mut {
slice/mod.rs:1114:macro_rules! iterator {
slice/mod.rs:1293:macro_rules! make_slice {
slice/mod.rs:1307:macro_rules! make_mut_slice {
slice/mod.rs:1330:macro_rules! search_while {
slice/mod.rs:2042:macro_rules! forward_iterator {
slice/mod.rs:2630:macro_rules! impl_marker_for {
slice/sort.rs:309:            macro_rules! left { () => { l.offset(*start_l as isize) } }
slice/sort.rs:310:            macro_rules! right { () => { r.offset(-(*start_r as isize) - 1) } }
array.rs:60:macro_rules! __impl_slice_eq1 {
array.rs:75:macro_rules! __impl_slice_eq2 {
array.rs:93:macro_rules! array_impls {
array.rs:223:macro_rules! array_impl_default {
internal_macros.rs:14:macro_rules! forward_ref_unop {
internal_macros.rs:34:macro_rules! forward_ref_binop {
marker.rs:359:macro_rules! impls{
hash/mod.rs:498:    macro_rules! impl_write {
hash/mod.rs:552:    macro_rules! impl_hash_tuple {
hash/mod.rs:572:    macro_rules! last_type {
hash/sip.rs:86:macro_rules! compress {
hash/sip.rs:106:macro_rules! load_int_le {
tests/hash/sip.rs:28:macro_rules! u8to64_le {
tests/char.rs:369:    macro_rules! assert_decode_utf8 {
tests/num/mod.rs:117:macro_rules! test_impl_from {
tests/num/mod.rs:215:macro_rules! test_impl_try_from_always_ok {
tests/num/mod.rs:266:macro_rules! test_impl_try_from_signed_to_unsigned_upper_ok {
tests/num/mod.rs:298:macro_rules! test_impl_try_from_unsigned_to_signed_upper_err {
tests/num/mod.rs:328:macro_rules! test_impl_try_from_same_sign_err {
tests/num/mod.rs:371:macro_rules! test_impl_try_from_signed_to_unsigned_err {
tests/num/mod.rs:403:macro_rules! test_float {
tests/num/uint_macros.rs:11:macro_rules! uint_module { ($T:ident, $T_i:ident) => (
tests/num/dec2flt/mod.rs:21:macro_rules! test_literal {
tests/num/int_macros.rs:11:macro_rules! int_module { ($T:ident, $T_i:ident) => (
tests/num/flt2dec/mod.rs:37:macro_rules! check_shortest {
tests/num/flt2dec/mod.rs:70:macro_rules! try_exact {
tests/num/flt2dec/mod.rs:81:macro_rules! try_fixed {
tests/num/flt2dec/mod.rs:206:macro_rules! check_exact {
tests/num/flt2dec/mod.rs:212:macro_rules! check_exact_one {
tests/num/flt2dec/estimator.rs:20:    macro_rules! assert_almost_eq {
sync/atomic.rs:922:macro_rules! atomic_int {
ops/arith.rs:104:macro_rules! add_impl {
ops/arith.rs:178:macro_rules! sub_impl {
ops/arith.rs:301:macro_rules! mul_impl {
ops/arith.rs:430:macro_rules! div_impl_integer {
ops/arith.rs:448:macro_rules! div_impl_float {
ops/arith.rs:510:macro_rules! rem_impl_integer {
ops/arith.rs:529:macro_rules! rem_impl_float {
ops/arith.rs:595:macro_rules! neg_impl_core {
ops/arith.rs:610:macro_rules! neg_impl_numeric {
ops/arith.rs:615:macro_rules! neg_impl_unsigned {
ops/arith.rs:669:macro_rules! add_assign_impl {
ops/arith.rs:726:macro_rules! sub_assign_impl {
ops/arith.rs:772:macro_rules! mul_assign_impl {
ops/arith.rs:818:macro_rules! div_assign_impl {
ops/arith.rs:863:macro_rules! rem_assign_impl {
ops/bit.rs:53:macro_rules! not_impl {
ops/bit.rs:137:macro_rules! bitand_impl {
ops/bit.rs:221:macro_rules! bitor_impl {
ops/bit.rs:308:macro_rules! bitxor_impl {
ops/bit.rs:391:macro_rules! shl_impl {
ops/bit.rs:408:macro_rules! shl_impl_all {
ops/bit.rs:495:macro_rules! shr_impl {
ops/bit.rs:512:macro_rules! shr_impl_all {
ops/bit.rs:607:macro_rules! bitand_assign_impl {
ops/bit.rs:652:macro_rules! bitor_assign_impl {
ops/bit.rs:697:macro_rules! bitxor_assign_impl {
ops/bit.rs:742:macro_rules! shl_assign_impl {
ops/bit.rs:755:macro_rules! shl_assign_impl_all {
ops/bit.rs:808:macro_rules! shr_assign_impl {
ops/bit.rs:821:macro_rules! shr_assign_impl_all {
cmp.rs:758:    macro_rules! partial_eq_impl {
cmp.rs:782:    macro_rules! eq_impl {
cmp.rs:791:    macro_rules! partial_ord_impl {
cmp.rs:834:    macro_rules! ord_impl {
ptr.rs:884:macro_rules! fnptr_impls_safety_abi {
ptr.rs:944:macro_rules! fnptr_impls_args {
tuple.rs:17:macro_rules! tuple_impls {
tuple.rs:94:macro_rules! lexical_ord {
tuple.rs:102:macro_rules! lexical_partial_cmp {
tuple.rs:112:macro_rules! lexical_cmp {
tuple.rs:122:macro_rules! last_type {
macros.rs:15:macro_rules! panic {
macros.rs:81:macro_rules! assert {
macros.rs:116:macro_rules! assert_eq {
macros.rs:164:macro_rules! assert_ne {
macros.rs:236:macro_rules! debug_assert {
macros.rs:261:macro_rules! debug_assert_eq {
macros.rs:286:macro_rules! debug_assert_ne {
macros.rs:343:macro_rules! try {
macros.rs:397:macro_rules! write {
macros.rs:442:macro_rules! writeln {
macros.rs:497:macro_rules! unreachable {
macros.rs:558:macro_rules! unimplemented {
macros.rs:565:/// These macros do not have any corresponding definition with a `macro_rules!`
macros.rs:580:    macro_rules! compile_error { ($msg:expr) => ({ /* compiler built-in */ }) }
macros.rs:590:    macro_rules! format_args { ($fmt:expr, $($args:tt)*) => ({
macros.rs:602:    macro_rules! env { ($name:expr) => ({ /* compiler built-in */ }) }
macros.rs:612:    macro_rules! option_env { ($name:expr) => ({ /* compiler built-in */ }) }
macros.rs:622:    macro_rules! concat_idents {
macros.rs:634:    macro_rules! concat { ($($e:expr),*) => ({ /* compiler built-in */ }) }
macros.rs:644:    macro_rules! line { () => ({ /* compiler built-in */ }) }
macros.rs:654:    macro_rules! column { () => ({ /* compiler built-in */ }) }
macros.rs:664:    macro_rules! file { () => ({ /* compiler built-in */ }) }
macros.rs:674:    macro_rules! stringify { ($t:tt) => ({ /* compiler built-in */ }) }
macros.rs:684:    macro_rules! include_str { ($file:expr) => ({ /* compiler built-in */ }) }
macros.rs:694:    macro_rules! include_bytes { ($file:expr) => ({ /* compiler built-in */ }) }
macros.rs:704:    macro_rules! module_path { () => ({ /* compiler built-in */ }) }
macros.rs:714:    macro_rules! cfg { ($($cfg:tt)*) => ({ /* compiler built-in */ }) }
macros.rs:724:    macro_rules! include { ($file:expr) => ({ /* compiler built-in */ }) }
char.rs:839:            macro_rules! first_byte {
char.rs:844:            macro_rules! continuation_byte {
fmt/mod.rs:1483:macro_rules! fmt_refs {
fmt/mod.rs:1638:macro_rules! peel {
fmt/mod.rs:1642:macro_rules! tuple {
fmt/mod.rs:1662:macro_rules! last_type {
fmt/float.rs:123:macro_rules! floating {
fmt/num.rs:36:macro_rules! doit {
fmt/num.rs:123:macro_rules! radix {
fmt/num.rs:146:macro_rules! int_base {
fmt/num.rs:157:macro_rules! debug {
fmt/num.rs:168:macro_rules! integer {
fmt/num.rs:197:macro_rules! impl_Display {
num/mod.rs:99:macro_rules! int_impl {
num/mod.rs:1261:macro_rules! uint_impl {
num/mod.rs:2471:macro_rules! from_str_radix_int_impl {
num/mod.rs:2507:macro_rules! same_sign_try_from_int_impl {
num/mod.rs:2539:macro_rules! cross_sign_from_int_impl {
num/mod.rs:2588:macro_rules! doit {
num/mod.rs:2722:macro_rules! impl_from {
num/uint_macros.rs:13:macro_rules! uint_module {
num/wrapping.rs:16:macro_rules! sh_impl_signed {
num/wrapping.rs:64:macro_rules! sh_impl_unsigned {
num/wrapping.rs:105:macro_rules! sh_impl_all {
num/wrapping.rs:124:macro_rules! wrapping_impl {
num/dec2flt/rawfp.rs:134:macro_rules! other_constants {
num/dec2flt/mod.rs:109:macro_rules! from_str_float_impl {
num/int_macros.rs:13:macro_rules! int_module {
num/bignum.rs:55:macro_rules! impl_full_ops {
num/bignum.rs:108:macro_rules! define_bignum {
```

**Comment by farodin91** [comment]
clean up
libstd
```
ascii.rs:1220:        macro_rules! test {
ascii.rs:1243:        macro_rules! test {
ascii.rs:1292:    macro_rules! assert_all {
ascii.rs:1313:    macro_rules! assert_none {
ffi/os_str.rs:623:macro_rules! impl_cmp {
os/raw.rs:92:    macro_rules! ok {
net/mod.rs:105:macro_rules! doit {
net/addr.rs:840:        macro_rules! try_opt {
net/udp.rs:708:    macro_rules! t {
net/tcp.rs:805:    macro_rules! t {
sync/mpsc/select.rs:375:    macro_rules! select {
fs.rs:1871:    macro_rules! check { ($e:expr) => (
fs.rs:1879:    macro_rules! error { ($e:expr, $s:expr) => (
fs.rs:1888:    macro_rules! error { ($e:expr, $s:expr) => (
time/mod.rs:445:    macro_rules! assert_almost_eq {
sys_common/mod.rs:99:macro_rules! rtabort {
sys_common/backtrace.rs:338:                    macro_rules! demangle {
sys_common/backtrace.rs:394:    macro_rules! t { ($a:expr, $b:expr) => ({
path.rs:2459:macro_rules! impl_cmp {
path.rs:2497:macro_rules! impl_cmp_os_str {
path.rs:2560:    macro_rules! t(
path.rs:3529:        macro_rules! tp(
path.rs:3623:        macro_rules! tp(
path.rs:3683:        macro_rules! tfn(
path.rs:3717:        macro_rules! tfe(
path.rs:3754:        macro_rules! t {
path.rs:3779:        macro_rules! tc(
num.rs:129:    macro_rules! test_is_power_of_two {
num.rs:150:    macro_rules! test_next_power_of_two {
num.rs:170:    macro_rules! test_checked_next_power_of_two {
num.rs:202:        macro_rules! assert_pow {
thread/local.rs:139:macro_rules! thread_local {
thread/local.rs:162:macro_rules! __thread_local_inner {
thread/local.rs:193:macro_rules! thread_local {
thread/local.rs:229:macro_rules! __thread_local_inner {
```
libcore
```
iter/range.rs:50:macro_rules! step_identical_methods {
iter/range.rs:74:macro_rules! step_impl_unsigned {
iter/range.rs:103:macro_rules! step_impl_signed {
iter/range.rs:145:macro_rules! step_impl_no_between {
iter/range.rs:178:macro_rules! range_exact_iter_impl {
iter/range.rs:185:macro_rules! range_incl_exact_iter_impl {
iter/range.rs:194:macro_rules! range_trusted_len_impl {
iter/range.rs:201:macro_rules! range_incl_trusted_len_impl {
iter/traits.rs:663:macro_rules! integer_sum_product {
iter/traits.rs:703:macro_rules! float_sum_product {
clone.rs:140:macro_rules! clone_impl {
default.rs:128:macro_rules! default_impl {
str/mod.rs:779:macro_rules! derive_pattern_clone {
str/mod.rs:830:macro_rules! generate_pattern_iterators {
str/mod.rs:1394:        macro_rules! err {
str/mod.rs:1403:        macro_rules! next { () => {{
str/pattern.rs:349:macro_rules! pattern_methods {
str/pattern.rs:377:macro_rules! searcher_methods {
slice/mod.rs:230:macro_rules! slice_offset {
slice/mod.rs:242:macro_rules! make_ref {
slice/mod.rs:255:macro_rules! make_ref_mut {
slice/mod.rs:1114:macro_rules! iterator {
slice/mod.rs:1293:macro_rules! make_slice {
slice/mod.rs:1307:macro_rules! make_mut_slice {
slice/mod.rs:1330:macro_rules! search_while {
slice/mod.rs:2042:macro_rules! forward_iterator {
slice/mod.rs:2630:macro_rules! impl_marker_for {
slice/sort.rs:309:            macro_rules! left { () => { l.offset(*start_l as isize) } }
slice/sort.rs:310:            macro_rules! right { () => { r.offset(-(*start_r as isize) - 1) } }
array.rs:60:macro_rules! __impl_slice_eq1 {
array.rs:75:macro_rules! __impl_slice_eq2 {
array.rs:93:macro_rules! array_impls {
array.rs:223:macro_rules! array_impl_default {
internal_macros.rs:14:macro_rules! forward_ref_unop {
internal_macros.rs:34:macro_rules! forward_ref_binop {
marker.rs:359:macro_rules! impls{
hash/mod.rs:498:    macro_rules! impl_write {
hash/mod.rs:552:    macro_rules! impl_hash_tuple {
hash/mod.rs:572:    macro_rules! last_type {
hash/sip.rs:86:macro_rules! compress {
hash/sip.rs:106:macro_rules! load_int_le {
sync/atomic.rs:922:macro_rules! atomic_int {
ops/arith.rs:104:macro_rules! add_impl {
ops/arith.rs:178:macro_rules! sub_impl {
ops/arith.rs:301:macro_rules! mul_impl {
ops/arith.rs:430:macro_rules! div_impl_integer {
ops/arith.rs:448:macro_rules! div_impl_float {
ops/arith.rs:510:macro_rules! rem_impl_integer {
ops/arith.rs:529:macro_rules! rem_impl_float {
ops/arith.rs:595:macro_rules! neg_impl_core {
ops/arith.rs:610:macro_rules! neg_impl_numeric {
ops/arith.rs:615:macro_rules! neg_impl_unsigned {
ops/arith.rs:669:macro_rules! add_assign_impl {
ops/arith.rs:726:macro_rules! sub_assign_impl {
ops/arith.rs:772:macro_rules! mul_assign_impl {
ops/arith.rs:818:macro_rules! div_assign_impl {
ops/arith.rs:863:macro_rules! rem_assign_impl {
ops/bit.rs:53:macro_rules! not_impl {
ops/bit.rs:137:macro_rules! bitand_impl {
ops/bit.rs:221:macro_rules! bitor_impl {
ops/bit.rs:308:macro_rules! bitxor_impl {
ops/bit.rs:391:macro_rules! shl_impl {
ops/bit.rs:408:macro_rules! shl_impl_all {
ops/bit.rs:495:macro_rules! shr_impl {
ops/bit.rs:512:macro_rules! shr_impl_all {
ops/bit.rs:607:macro_rules! bitand_assign_impl {
ops/bit.rs:652:macro_rules! bitor_assign_impl {
ops/bit.rs:697:macro_rules! bitxor_assign_impl {
ops/bit.rs:742:macro_rules! shl_assign_impl {
ops/bit.rs:755:macro_rules! shl_assign_impl_all {
ops/bit.rs:808:macro_rules! shr_assign_impl {
ops/bit.rs:821:macro_rules! shr_assign_impl_all {
cmp.rs:758:    macro_rules! partial_eq_impl {
cmp.rs:782:    macro_rules! eq_impl {
cmp.rs:791:    macro_rules! partial_ord_impl {
cmp.rs:834:    macro_rules! ord_impl {
ptr.rs:884:macro_rules! fnptr_impls_safety_abi {
ptr.rs:944:macro_rules! fnptr_impls_args {
tuple.rs:17:macro_rules! tuple_impls {
tuple.rs:94:macro_rules! lexical_ord {
tuple.rs:102:macro_rules! lexical_partial_cmp {
tuple.rs:112:macro_rules! lexical_cmp {
tuple.rs:122:macro_rules! last_type {
char.rs:839:            macro_rules! first_byte {
char.rs:844:            macro_rules! continuation_byte {
fmt/mod.rs:1483:macro_rules! fmt_refs {
fmt/mod.rs:1638:macro_rules! peel {
fmt/mod.rs:1642:macro_rules! tuple {
fmt/mod.rs:1662:macro_rules! last_type {
fmt/float.rs:123:macro_rules! floating {
fmt/num.rs:36:macro_rules! doit {
fmt/num.rs:123:macro_rules! radix {
fmt/num.rs:146:macro_rules! int_base {
fmt/num.rs:157:macro_rules! debug {
fmt/num.rs:168:macro_rules! integer {
fmt/num.rs:197:macro_rules! impl_Display {
num/mod.rs:99:macro_rules! int_impl {
num/mod.rs:1261:macro_rules! uint_impl {
num/mod.rs:2471:macro_rules! from_str_radix_int_impl {
num/mod.rs:2507:macro_rules! same_sign_try_from_int_impl {
num/mod.rs:2539:macro_rules! cross_sign_from_int_impl {
num/mod.rs:2588:macro_rules! doit {
num/mod.rs:2722:macro_rules! impl_from {
num/uint_macros.rs:13:macro_rules! uint_module {
num/wrapping.rs:16:macro_rules! sh_impl_signed {
num/wrapping.rs:64:macro_rules! sh_impl_unsigned {
num/wrapping.rs:105:macro_rules! sh_impl_all {
num/wrapping.rs:124:macro_rules! wrapping_impl {
num/dec2flt/rawfp.rs:134:macro_rules! other_constants {
num/dec2flt/mod.rs:109:macro_rules! from_str_float_impl {
num/int_macros.rs:13:macro_rules! int_module {
num/bignum.rs:55:macro_rules! impl_full_ops {
num/bignum.rs:108:macro_rules! define_bignum {
```

**Comment by vlad20012** [comment]
Fix was merged with #1516
